src/components/Identity/createComponent.js

```diff
@@ -61,14 +55,23 @@ export default (processIdSyncs, config, logger, consent, eventManager) => {
         // new visitor).
         deferredForEcid = defer();
         payload.expectResponse();
```

Comment:
So we are expecting a response because we need to get a cookie back?  Does /collect never set cookies, or is it that because we use sendBeacon when calling /collect we cannot get cookies back?  It seems like there are two concerns here 1. whether or not we expect a response (and therefore whether or not to call /interact vs /collect) and 2. whether or not we expect a cookie in return (and therefore whether to use fetch vs sendBeacon).

Reply:
Good questions.

> So we are expecting a response because we need to get a cookie back?

Yes.

> Does /collect never set cookies, or is it that because we use sendBeacon when calling /collect we cannot get cookies back?

There are a couple cases where first-party cookies can't be set (either by Konductor or by Konductor telling Alloy to write a cookie):

1. When Alloy calls `/collect` when Konductor is on a different domain (e.g., edge.adobedc.net), Konductor can't set cookies on its own and won't send a response body so it can't tell Alloy to write a cookie.
2. When Alloy calls `/collect` using `sendBeacon`.

I brought this up with Rares and he said if we don't have an identity cookie, we need to hit `interact` instead of `collect`. It's an interesting scenario though, because what if Konductor needs to set cookies other than the identity cookie. Alloy isn't aware of all the cookies Konductor might need to set, and really shouldn't be aware. I brought this up with Rares too and he said the best decision we can make right now is to just focus on the identity cookie and hit `interact` endpoint if the identity cookie doesn't exist. This is sufficient for the cookies we have currently. He's not 100% sure this will be sufficient for the future, just 99%.

If you have ideas on how we could improve this, let us know. I can add some more comments around this.

Reply:
Thank you for the clarification.  My thought here was maybe we should have an "expectCookie()" call on the payload instead of overloading the meaning of expectResponse().  I can see it either way though because you could think of a set-cookie header as part of the response.

Reply:
> you could think of a set-cookie header as part of the response

Yes, partly this, and also we don't know if the cookie will come as a `set-cookie` header or in the response body (as a `state:store` handle) without looking to see if we're making a request to Konductor through a third-party domain. I thought it would be simpler to just stick with the concept we already have (`expectResponse()`) for both cases.